README.md

```diff
@@ -1,348 +1,143 @@
-if it were the last, and each onLoad() should check to see if it's really a
-reload. But that's a lot simpler than the usual gymnastics.
-
-### Singletons Where You Want Them
-
-Under the hood, scopes take advantage of one of Dagger's most interesting
-features, the ability for an object graph to spawn a child graph (see
-[ObjectGraph#plus][ogplus]). Singletons and other bindings defined in a parent
-graph can be injected by all of its offspring, but the parent graph itself is
-not modified by them. In practice, this means that a `@Singleton FooService`
-defined in the root scope is available for injection to all parts of the app.
-Any singletons defined in an activity's scope, say an `@Singleton FooEditor`,
-can be injected by the activity and any of its views, but is not accessible to
-the rest of the application.
-
-When a scope is destroyed it drops all references to its own object graph and
-makes its children do the same, freeing up everything to be GC'd. Each portion
-of the app has the convenience of global singletons, with no concerns that
-precious RAM is being consumed by things the user doesn't care about at the
-moment.
-
-Another example: suppose you have a set of screens that slide in and out next
-to a fixed portion of the screen, like a sidebar or an activity, and each
-screen needs to share in the management of that common region. The parent view
-that owns the common ground can be tied to a parent scope, and starts and
-stops child scopes for each child view. Children can simply inject the
-presenter that drives the shared region. They don't need to know how far
-up the view chain that action bar lives, or to somehow find their way to the
-activity. It's simply available.
-
-## Less Talk More Code
-
-Here's how you might actually write this stuff.
-
-This example presumes an activity is in charge of displaying subscreens and that
-you're using Flow's [Layouts][layouts] utility to handle view creation.
-
-```java
-/**
- * @param nextScreen blueprint of the screen to show, must have
- * {@literal @}Layout annotation.
- */
-void showScreen(Blueprint nextScreen) {
-  MortarScope activityScope = Mortar.getMortarScope(this);
-
-  View currentView = getCurrentView();
-  if (currentView != null) {
-    activityScope.destroyChild(Mortar.getMortarScope(currentView.getContext()));
+  @Override public void onCreate() {
+    super.onCreate();
+    rootScope = MortarScope.buildRootScope().build();
   }
 
-  MortarScope newScope = activityScope.requireChild(nextScreen);
-  Context newContext = newScope.createContext(this);
-  View screenView = Layouts.createView(newContext, nextScreen);
-
-  setContentView(screenView);
-}
+  @Override public Object getSystemService(String name) {
+    Object serviceFromMortar = rootScope.getService(name);
+    if (serviceFromMortar != null) return serviceFromMortar;
 
-View getCurrentView() {
-  return ((ViewGroup)findViewById(android.R.id.content)).getChildAt(0);
+    return super.getSystemService(name);
```

Comment:
Is this the right pattern?  It makes it impossible to hide a service by shadowing it with null, a capability that you do have with Context.